service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -151,20 +187,33 @@ public PrometheusBasedResourceLimitChecks(final WebClient webClient,
         } else {
             final String queryParams = String.format("sum(%s{tenant=\"%s\"})", CONNECTIONS_METRIC_NAME,
                     tenant.getTenantId());
-            return executeQuery(queryParams)
+            return executeQuery(queryParams, span)
                     .map(currentConnections -> {
                         if (currentConnections < maxConnections) {
+                            span.finish();
```

Comment:
how about logging the number of current connections vs the max connections here as well?

Reply:
The logging as mentioned above has been added.